backend/src/main/java/org/sefglobal/core/academix/service/ItemService.java

```diff
@@ -81,31 +80,35 @@ public Item addItem(List<Long> subCategoryIds, Item item) throws ResourceNotFoun
      *
      * @param id   which is the {@link Item} to be updated
      * @param item which holds the the updated data
-     * @return {@code true} if {@link Item} gets updated
+     * @return the updated {@link Item}
      *
      * @throws ResourceNotFoundException is thrown if the requesting {@link Item} doesn't exist
+     * @throws ResourceNotFoundException is thrown if the requesting {@link SubCategory} doesn't
+     * exist
      */
-    public boolean updateItem(long id, Item item)
+    public Item updateItem(long id, Item item)
             throws ResourceNotFoundException {
-        boolean isUpdated = itemRepository
-                .findById(id)
-                .map(updatableItem -> {
-                    item.getTranslations().forEach(updatedTranslation -> {
-                        itemTranslationRepository
-                                .findById(new ItemTranslationId(updatableItem, updatedTranslation.getLanguage()))
-                                .ifPresent(updatableTranslation ->
-                                                   updatableTranslation.setName(updatedTranslation.getName()));
-                        updatableItem.addTranslation(updatedTranslation);
-                    });
-                    return itemRepository.save(updatableItem);
-                })
-                .isPresent();
-        if (!isUpdated) {
-            String msg = "Error, Item with id: " + id + " cannot be updated. Item doesn't exist.";
+        if (!itemRepository.existsById(id)) {
+            String msg = "Error, Item with id: " + id + " cannot be updated. Item " +
+                    "doesn't exist.";
             log.error(msg);
             throw new ResourceNotFoundException(msg);
+        } else {
+            List<Long> subCategoryIds = new ArrayList<>();
+            List<SubCategory> subCategories = item.getSubCategories();
+            for (SubCategory subCategory: subCategories){
+                subCategoryIds.add(subCategory.getId());
+            }
+            List<SubCategory> existingSubCategories = subCategoryRepository.findAllById(subCategoryIds);
+            if (subCategoryIds.size() > existingSubCategories.size()) {
```

Comment:
You can remove the loop
```suggestion
            List<SubCategory> subCategories = item.getSubCategories();
            List<SubCategory> existingSubCategories = subCategoryRepository.findAllById(subCategoryIds);
            if (subCategories.size() > existingSubCategories.size()) {
```

Reply:
But to get the list of subcategories, we need the list of subcategory Ids. 
`List<Long> subCategories`

Then the IDs taken from the request body are added to the List
Followed by the SubCategories relevant to each id are loaded into the `subCategories` List